scalactic-macro/src/main/scala/org/scalactic/source/PositionMacro.scala

```diff
@@ -1,65 +1,75 @@
-/*
- * Copyright 2001-2016 Artima, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package org.scalactic.source
 
-import org.scalactic.Resources
 
+//import scala.reflect.macros.blackbox.Context
 import scala.reflect.macros.Context
+import org.scalactic.Resources
 
 /**
  * Helper class for Position macro. (Will be removed from the public API if possible in a subsequent 3.0.0-RCx release.)
  */
-object PositionMacro {
+object PositionMacro {   
+  private object sourceCompatHack {
+    object internal { object decorators {} }
+  }
+  import sourceCompatHack._
+
+  private class PositionMacroImpl(val universe: scala.reflect.api.Universe) {
+
+    private val PositionModule = universe.rootMirror.staticModule("org.scalactic.source.Position")
+    private val PositionClass = universe.rootMirror.staticClass("org.scalactic.source.Position")
+    private val Position_apply = PositionModule.typeSignature.declaration(universe.newTermName("apply"))
+
+    private[scalactic] lazy val showScalacticFillFilePathnames: Boolean = {
+      val value = System.getenv("SCALACTIC_FILL_FILE_PATHNAMES")
+      value != null && value == "yes"
+    }
```

Comment:
`showScalacticFillFilePathnames` could still be a member of `PositionMacro`.

Reply:
This is a good idea, thanks!